lib/memory_analyzer/heap_analyzer/parser.rb

```diff
@@ -93,6 +95,15 @@ def string_pool
       def string_pool_intern(string)
         string_pool.add(string)[string]
       end
+
+      def io_klass
+        if File.extname(file) == ".gz"
+          require 'zlib'
+          Zlib::GzipReader
```

Comment:
I think there's a way to read the magic number of the file and use that...let me find out.

Reply:
I can look as well, I was just going with this for simplicity, but I think we should probably do both.

Reply:
Yeah, do the following:

```
if File.read(file, 2, :mode => "rb") == "\x1F\x8B".force_encoding("ASCII-8BIT")
  require 'zlib'
  Zlib::GzipReader
else
  File
end
```

Reply:
@Fryguy Does it make sense to do both?

```ruby
if File.read(file, 2, :mode => "rb") == GZ_MAGIC_NUMBER
```

Also, isn't there another character of `08` to that magic number?

https://www.garykessler.net/library/file_sigs.html

Reply:
can't hurt to do both.

Reply:
Looks like `Zlib` actually does some of this already:

https://github.com/ruby/ruby/blob/de5e347/ext/zlib/zlib.c#L2135-L2136

So might be able to make use of that.  Also, that further proves your point about the `08` being not necessary, thanks.